docs/csharp/language-reference/operators/lambda-expressions.md

```diff
@@ -204,12 +204,78 @@ customers.Where(c => c.City == "London");
+
+```csharp
+object parse = (string s) => int.Parse(s);   // Func<string, int>
+Delegate parse = (string s) => int.Parse(s); // Func<string, int>
+```
+
+Method groups (that is, method names without argument lists) with exactly one overload have a natural type:
+
+```csharp
+var read = Console.Read; // Just one overload; Func<int> inferred
+var write = Console.Write; // ERROR: Multiple overloads, can't choose
+```
+
+If you assign a lambda expression to <xref:System.Linq.Expressions.LambdaExpression?displayProperty=nameWithType>, or <xref:System.Linq.Expressions.Expression?displayProperty=nameWithType>, and the lambda has a natural delegate type, the expression has a natural type of <xref:System.Linq.Expressions.Expression%601?displayProperty=nameWithType>, with the natural delegate type used as the argument for the type parameter:
+
+```csharp
+LambdaExpression parseExpr = (string s) => int.Parse(s); // Expression<Func<string, int>>
+Expression parseExpr = (string s) => int.Parse(s);       // Expression<Func<string, int>>
+```
+
+Many lambda expressions won't have a natural type. Consider the following declaration:
+
+```csharp
+var parse = s => int.Parse(s); // ERROR: Not enough type info in the lambda
+```
+
+The compiler can't infer a parameter type for `s`. When the compiler can't infer a natural type, you must declare the type:
+
+```csharp
+Func<string, int> parse = s => int.Parse(s);
+```
+
+## Declared return type
+
+Typically, the return type of the lambda expression is obvious and inferred. For some expressions, that wouldn't work:
+
+``` c#
+var choose = (bool b) => b ? 1 : "two"; // ERROR: Can't infer return type
+```
+
+Beginning with C# 10, you can specify the return type on a lambda expression before the parameters. When you specify an explicit return type, the parameters must be parenthesized, so that it's not too confusing to the compiler or other developers:
+
+``` c#
+var choose = object (bool b) => b ? 1 : "two"; // Func<bool, object>
+```
+
+## Attributes
+
+Beginning with C# 10, you can apply attributes to lambda expressions. The attributes are added before the parameter declaration. The lambda expression's parameter list must be parenthesized when there are attributes:
+
+``` c#
+Func<string, int> parse = [Example(1)] (s) => int.Parse(s);
+var choose = [Example(2)][Example(3)] object (bool b) => b ? 1 : "two";
+```
+
+You can apply any attribute that is valid on <xref:System.AttributeTargets.Method?displayProperty=nameWithType>.
+
+Lambda expressions are invoked differently than methods and local functions. Attributes don't have any effect when the lambda expression is invoked. Attributes on lambda expressions are useful for code analysis, and can be discovered via reflection.
```

Comment:
I'm not sure I entirely follow what you're implying here. If attributes do not have any effect when the lambda expression is invoked, what attributes should be avoided? For example, with Minimal API could I add an attribute to an expression that defines `RequiresScope`? Would that work?

Reply:
Added clarification to that. 

I think the answer is no. Unless the ASP.NET team is using reflection to see the attributes on the lambda expression, the library won't have that information.